src/Validation/Validator.php

```diff
@@ -1941,6 +1943,54 @@ public function regex($field, $regex, $message = null, $when = null)
         ]);
     }
 
+    /**
+     * Gets the required message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getRequiredMessage($field)
+    {
+        $defaultMessage = static::DEFAULT_REQUIRED_MESSAGE;
+        if ($this->_useI18n) {
+            $defaultMessage = __d('cake', $defaultMessage);
+        }
+
+        return isset($this->_presenceMessages[$field])
+            ? $this->_presenceMessages[$field]
+            : $defaultMessage;
+    }
+
+    /**
+     * Gets the notEmpty message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getNotEmptyMessage($field)
+    {
+        $defaultMessage = static::DEFAULT_EMPTY_MESSAGE;
+        if ($this->_useI18n) {
+            $defaultMessage = __d('cake', $defaultMessage);
+        }
+
+        $notBlankMessage = null;
+        if (isset($this->_fields[$field])) {
+            foreach ($this->_fields[$field] as $rule) {
+                if ($rule->get('rule') === 'notBlank') {
+                    $notBlankMessage = $rule->get('message');
+                    break;
```

Comment:
If this also checked the message truthiness, couldn't you return here and save some logic below?